python_sdk/infrahub_sdk/ctl/cli_commands.py

```diff
@@ -296,6 +298,106 @@ def transform(
+) -> None:
+    """Export Python protocols corresponding to a schema."""
+
+    def _jinja2_filter_inheritance(value: dict[str, Any]) -> str:
+        inherit_from: list[str] = value.get("inherit_from", [])
+
+        if not inherit_from:
+            return "CoreNode"
+        return ", ".join(inherit_from)
+
+    def _jinja2_filter_render_attribute(value: AttributeSchema) -> str:
+        attribute_kind_map = {
+            "boolean": "bool",
+            "datetime": "datetime",
+            "dropdown": "str",
+            "hashedpassword": "str",
+            "iphost": "str",
+            "ipnetwork": "str",
+            "json": "dict",
+            "list": "list",
+            "number": "int",
+            "password": "str",
+            "text": "str",
+            "textarea": "str",
+            "url": "str",
+        }
+
+        name = value.name
+        kind = value.kind
+
+        return f"{name}: {attribute_kind_map[kind.lower()]}"
+
+    def _jinja2_filter_render_relationship(value: RelationshipSchema, sync: bool = False) -> str:
+        name = value.name
+        cardinality = value.cardinality
+
+        type_ = "RelatedNode"
+        if cardinality == "many":
+            type_ = "RelationshipManager"
+
+        if sync:
+            type_ += "Sync"
+
+        return f"{name}: {type_}"
+
+    def _sort_and_filter_models(
+        models: dict[str, Union[AttributeSchema, RelationshipSchema]], filters: Optional[list[str, str]] = None
+    ) -> list[dict[str, Any]]:
+        if filters is None:
+            filters = ["CoreNode"]
+
+        filtered: list[Union[AttributeSchema, RelationshipSchema]] = []
+        for name, model in models.items():
+            if name in filters:
+                continue
+            filtered.append(model)
+
+        return sorted(filtered, key=lambda k: k.name)
+
+    client = initialize_client_sync()
+    current_schema = client.schema.all(branch=branch)
+
+    generics: Dict[str, GenericSchema] = {}
+    nodes: Dict[str, NodeSchema] = {}
+
+    for name, schema_type in current_schema.items():
+        if isinstance(schema_type, GenericSchema):
+            generics[name] = schema_type
+        if isinstance(schema_type, NodeSchema):
+            nodes[name] = schema_type
+
+    jinja2_env = jinja2.Environment(loader=jinja2.BaseLoader, trim_blocks=True, lstrip_blocks=True)
+    jinja2_env.filters["inheritance"] = _jinja2_filter_inheritance
+    jinja2_env.filters["render_attribute"] = _jinja2_filter_render_attribute
+    jinja2_env.filters["render_relationship"] = _jinja2_filter_render_relationship
+
+    template = jinja2_env.from_string(PROTOCOLS_TEMPLATE)
+    rendered = template.render(
+        generics=_sort_and_filter_models(generics), nodes=_sort_and_filter_models(nodes), sync=False
+    )
+    rendered_sync = template.render(
+        generics=_sort_and_filter_models(generics), nodes=_sort_and_filter_models(nodes), sync=True
```

Comment:
you could assign the results of `_sort_and_filter_models` to variables so that you don't need to run it twice for sync and async versions